File: frontend/lib/features/explore/data/models/problem_model.dart
import '../../domain/entities/problem.dart';

class ProblemModel extends Problem {
  const ProblemModel({
    required super.id,
    required super.name,
    required super.category,
    required super.description,
    required super.severity,
    required super.treatmentDifficulty,
    required super.commonCauses,
    required super.solutions,
    required super.prevention,
    required super.affectedPlants,
    super.icon,
    super.color,
  });

  factory ProblemModel.fromJson(Map<String, dynamic> json) {
    return ProblemModel(
      id: json['id'] as String,
      name: json['name'] as String,
      category: json['category'] as String,
      description: json['description'] as String,
      severity: json['severity'] as String,
      treatmentDifficulty: json['treatmentDifficulty'] as String,
      commonCauses: (json['commonCauses'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      solutions: (json['solutions'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      prevention: json['prevention'] as String,
      affectedPlants: (json['affectedPlants'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      icon: json['icon'] as String?,
      color: json['color'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'category': category,
      'description': description,
      'severity': severity,
      'treatmentDifficulty': treatmentDifficulty,
      'commonCauses': commonCauses,
      'solutions': solutions,
      'prevention': prevention,
      'affectedPlants': affectedPlants,
      'icon': icon,
      'color': color,
    };
  }
}

File: backend/src/models/Problem.model.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IProblem extends Document {
  name: string;
  category: string;
  description: string;
  severity: 'Mild' | 'Moderate' | 'Severe';
  treatmentDifficulty: 'Easy' | 'Moderate' | 'Difficult';
  commonCauses: string[];
  solutions: string[];
  prevention: string;
  affectedPlants: string[];
  icon?: string;
  color?: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const ProblemSchema = new Schema<IProblem>(
  {
    name: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    category: {
      type: String,
      required: true,
      enum: ['Pests', 'Diseases', 'Environmental', 'Nutrition', 'Watering'],
      trim: true,
    },
    description: {
      type: String,
      required: true,
      trim: true,
    },
    severity: {
      type: String,
      required: true,
      enum: ['Mild', 'Moderate', 'Severe'],
      default: 'Moderate',
    },
    treatmentDifficulty: {
      type: String,
      required: true,
      enum: ['Easy', 'Moderate', 'Difficult'],
      default: 'Moderate',
    },
    commonCauses: {
      type: [String],
      default: [],
    },
    solutions: {
      type: [String],
      required: true,
      default: [],
    },
    prevention: {
      type: String,
      required: true,
      trim: true,
    },
    affectedPlants: {
      type: [String],
      default: [],
    },
    icon: {
      type: String,
      trim: true,
    },
    color: {
      type: String,
      trim: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

// Indexes
ProblemSchema.index({ category: 1, isActive: 1 });
ProblemSchema.index({ name: 'text', description: 'text' });

export const Problem = mongoose.model<IProblem>('Problem', ProblemSchema);
